nio/client/async_client.py

```diff
@@ -1242,11 +1243,14 @@ def add_response_callback(
         import_keys = partial(self.olm.import_keys_static, infile, passphrase)
         sessions = await loop.run_in_executor(None, import_keys)
 
+        to_save = []
+
         for session in sessions:
-            # This could be improved by writing everything to db at once at
-            # the end
             if self.olm.inbound_group_store.add(session):
-                self.store.save_inbound_group_session(session)
+                to_save.append(session)
+
+        save = partial(self.store.save_inbound_group_sessions, *to_save)
+        await loop.run_in_executor(None, save)
```

Comment:
I think this is the reason why you might see the `Database is locked error`, or was that even before you made your changes?

We should really finish up the threaded database support instead of putting a single write into a threaded executor. That way all writes would happen inside a writer thread and we wouldn't need to worry about database locked issues.